sct-commons/src/test/java/org/lfenergy/compas/sct/commons/scl/SclRootAdapterTest.java

```diff
@@ -64,4 +70,58 @@ void addPrivate() throws Exception {
         assertEquals(1, sclRootAdapter.getCurrentElem().getPrivate().size());
         assertIsMarshallable(scd);
     }
+
+    @Test
+    void getIEDAdapterByName_should_return_ied(){
+        // Given
+        SCL scl = new SCL();
+        scl.setHeader(new THeader());
+        TIED ied = new TIED();
+        ied.setName("IED_NAME");
+        scl.getIED().add(ied);
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scl);
+        // When
+        IEDAdapter resultIed = sclRootAdapter.getIEDAdapterByName("IED_NAME");
+        // Then
+        assertThat(resultIed.getName()).isEqualTo("IED_NAME");
+    }
+
+    @Test
+    void getIEDAdapterByName_should_throw_exception(){
+        // Given
+        SCL scl = new SCL();
+        scl.setHeader(new THeader());
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scl);
+        // When & Then
+        assertThatThrownBy(() -> sclRootAdapter.getIEDAdapterByName("IED_NAME"))
+            .isInstanceOf(ScdException.class);
+    }
+
+    @Test
+    void findIEDAdapterByName_should_return_ied(){
+        // Given
+        SCL scl = new SCL();
+        scl.setHeader(new THeader());
+        TIED ied = new TIED();
+        ied.setName("IED_NAME");
+        scl.getIED().add(ied);
+        SclRootAdapter sclRootAdapter = new SclRootAdapter(scl);
```

Comment:
This part is duplicated with lines 77-82, why not introduce a method like givenSclRootAdapter ?